libcontainer/configs/config_test.go

```diff
@@ -160,18 +161,53 @@ func TestCommandHookRun(t *testing.T) {
 		Pid:     1,
 		Bundle:  "/bundle",
 	}
-	timeout := time.Second
+
+	stateJson, err := json.Marshal(state)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	verifyCommandTemplate := `#!/bin/sh
+if [ "$1" != "testarg" ]; then
+	echo "Bad value for $1. Expected 'testarg', found '$1'"
+	exit 1
+fi
+if [ -z "$FOO" ] || [ "$FOO" != BAR ]; then
+	echo "Bad value for FOO. Expected 'BAR', found '$FOO'"
+	exit 1
+fi
+expectedJson=%q
+read JSON
+if [ "$JSON" != "$expectedJson" ]; then
+	echo "Bad JSON received. Expected '$expectedJson', found '$JSON'"
+	exit 1
+fi
+exit 0
+	`
+
+	verifyCommand := fmt.Sprintf(verifyCommandTemplate, stateJson)
+
+	tmpFile, err := ioutil.TempFile("", "runcTest")
+	if err != nil {
+		t.Fatalf("Failed to create tmp file: %v", err)
+	}
+	defer os.Remove(tmpFile.Name())
+	if _, err := tmpFile.Write([]byte(verifyCommand)); err != nil {
+		t.Fatalf("Failed to write tmp file: %v", err)
+	}
+	if err := tmpFile.Chmod(0777); err != nil {
```

Comment:
`0777` looks overly permissive. `0755` should be good enough and not overly restrictive, although even `0700` should work just fine.

Reply:
Also, there is no need to create a temp file, you can just run `sh -` and provide the script to stdin. Maybe this will simplify the code a bit

Reply:
I'm not sure how I could pass the script to stdin. The hook already receives the state using stdin. I could also use `sh -c 'script'` to avoid creating a temp file but passing the parameters gets tricky. I think it's more clear to create a temp script and directly execute it. 

I update the code to use 0700 and simplified it a bit.